src/tests/dlopen-tests.c

```diff
@@ -154,16 +155,84 @@ static bool recursive_dlopen(const char **name, int round, char **errmsg)
     return ok;
 }
 
+static int file_so_filter(const struct dirent *ent)
```

Comment:
Just a nitpick -- if function returns values 0 and 1 and it is answer to question of type yes/no, isn't bool better than int?